metaserver/src/main/scala/scala/meta/languageserver/SymbolIndexer.scala

```diff
@@ -29,19 +30,25 @@ class SymbolIndexer(
 
   def documentSymbols(
       path: RelativePath
-  ): Seq[(Position.Range, Denotation)] =
+  ): Seq[SymbolInformation] = {
     for {
       document <- Option(documents.get(path)).toList
       _ <- isFreshSemanticdb(path, document).toList
-      ResolvedName(pos: Position.Range, symbol: Symbol.Global, true) <- document.names
+      ResolvedName(position: Position.Range, symbol: Symbol.Global, true) <- document.names
       denotation <- Option(denotations.get(symbol))
       if ! {
         import denotation._
         isPrimaryCtor ||
         isTypeParam ||
         isParam
       } // not interesting for this service
-    } yield pos -> denotation
+    } yield SymbolInformation(
+      denotation.name,
+      denotation.symbolKind,
+      path.toAbsolute.toLocation(position),
+      Option(denotations.get(symbol.owner)).map(_.name)
```

Comment:
How about using `symbol.owner.syntax`? `.owner` is likely a `Symbol.Global` which has a `.signature.name` we can use

Reply:
See https://github.com/scalameta/language-server/issues/33#issuecomment-343774428.